pkg/smbios/table_type.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2016-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package smbios
+
+// TableType specifies the DMI type of the table.
+// Types are defined in DMTF DSP0134.
+type TableType uint8
+
+// Supported table types.
+const (
+	TableTypeBIOSInformation   TableType = 0
+	TableTypeSystemInformation           = 1
+	TableTypeInactive                    = 126
+	TableTypeEndOfTable                  = 127
+)
+
+func (t TableType) String() string {
+	switch t {
+	case TableTypeBIOSInformation:
+		return "BIOS Information"
+	case TableTypeSystemInformation:
+		return "System Information"
+	case TableTypeInactive:
+		return "Inactive"
+	case TableTypeEndOfTable:
+		return "End Of Table"
+	default:
+		if t >= 0x80 {
+			return "OEM-specific Type"
```

Comment:
return fmt.Sprintf("OEM-specific type (%#x)", t)

Better debugging

Reply:
i agree in principle but this would introduce diffs with existing dmidecode.
in practice, the line just before contains the type number (from header serialization), so the type of the table is clear to the user:

```
Handle 0x001e, DMI type 131, 64 bytes
OEM-specific Type
...
```